### ISSUE TITLE
error messages

### ISSUE THREAD

**Issue by r-mustafa** [roshchupkin/hase#2]
Hi,

I am trying to run analysis using HASE on a cluster but there was an error message:

/$HOME/hase-master/hdgwas/data.py:798: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
  self._data._data=df[df.columns[1:]].as_matrix()
Traceback (most recent call last):
  File "/$HOME/hase-master/hase.py", line 500, in <module>
    gen[i].start(j,hdf5=args.hdf5, study_name=args.study_name[i], ID=False)
  File "/$HOME/hase-master/hdgwas/data.py", line 1160, in start
    self.folder=Folder(path)
  File "/$HOME/hase-master/hdgwas/data.py", line 588, in __init__
    self._scan()
  File "/$HOME/hase-master/hdgwas/data.py", line 604, in _scan
    raise ValueError('There are different data format {} in {} folder'.format(set(exts) ,self.path))
ValueError: There are different data format set(['', '.txt']) in /$HOME/from_Sajjad/chr1/ folder

FYI, I am using Python 2.7.15 on my library, and the files in the genotype folder of chr1 folder already contained h5 formatted genotype file. 

Could you please help me with this?

**Comment by r-mustafa** [comment]
Hi,

The previous problem has been sorted. 
Fyi, I edited the data.py on line 798 by changing as.matrix into .values as it has been suggested by the previous error messages. 

Now it seems that my script is able to run and generate the output files. The problem is that the files are all empty. The error message file is empty, and the output file contains the following messages:

_Collecting package metadata: ...working... done
Solving environment: ...working... done

# All requested packages already installed.

WARNING!!! You did not set threshold for t-value, all results will be saved
Namespace(cluster='n', covariates='/rds/general/user/rm2715/home/hase_aw/covar/', derivatives=None, effect_intercept=False, encoded=None, genotype=['/rds/general/project/medbio-epiukb-archive-2018/live/projects/genomics_of_lipidomics/hase_affy/conversion/chr1/'], hdf5=True, id=False, interaction=None, intercept='y', maf=0.005, mapper=None, mapper_chunk=None, mode='regression', node=None, np=True, out='/rds/general/user/rm2715/home/hase_aw/results/chr1/', pd_full=False, permute_ph=False, ph_id_exc=None, ph_id_inc=None, phenotype=['/rds/general/user/rm2715/home/hase_aw/phenotype/'], protocol=None, ref_name='1000Gp1v3_ref', snp_id_exc=None, snp_id_inc=None, study_name=['affy_airwave_chr_one'], thr=0.0, vcf=False)
Numpy linked to lapack_opt_info
START regression mode...
reading file phenotype.txt
There are 1967 ids and 886 columns 
reading file covar.txt
There are 1967 ids and 12 columns 
Reading ID from probes....
There are 3908 ids
There are 0 common ids
Namespace(df=None, out='/rds/general/user/rm2715/home/hase_aw/results/chr1/', r='/rds/general/user/rm2715/home/hase_aw/results/chr1')
DF is not defined. Forced to use z_score statistics!

============================================

        Job resource usage summary 

                 Memory (GB)    NCPUs
 Requested  :       100             2
 Used       :         0 (peak)   0.61 (ave)

============================================_

Could you please help me with this?

**Comment by r-mustafa** [comment]
Hi,

The previous problem has been sorted.
Fyi, I edited the data.py on line 798 by changing as.matrix into .values as it has been suggested by the previous error messages.

Now it seems that my script is able to run and generate the output files. The problem is that the files are all empty. The error message file is empty, and the output file contains the following messages:

_Collecting package metadata: ...working... done
Solving environment: ...working... done
# All requested packages already installed.

WARNING!!! You did not set threshold for t-value, all results will be saved
Namespace(cluster='n', covariates='//hase_aw/covar/', derivatives=None, effect_intercept=False, encoded=None, genotype=['//hase_affy/conversion/chr1/'], hdf5=True, id=False, interaction=None, intercept='y', maf=0.005, mapper=None, mapper_chunk=None, mode='regression', node=None, np=True, out='/r/hase_aw/results/chr1/', pd_full=False, permute_ph=False, ph_id_exc=None, ph_id_inc=None, phenotype=['//hase_aw/phenotype/'], protocol=None, ref_name='1000Gp1v3_ref', snp_id_exc=None, snp_id_inc=None, study_name=['affy_airwave_chr_one'], thr=0.0, vcf=False)
Numpy linked to lapack_opt_info
START regression mode...
reading file phenotype.txt
There are 1967 ids and 886 columns
reading file covar.txt
There are 1967 ids and 12 columns
Reading ID from probes....
There are 3908 ids
There are 0 common ids
Namespace(df=None, out='//hase_aw/results/chr1/', r='//hase_aw/results/chr1')
DF is not defined. Forced to use z_score statistics!

============================================

```
    Job resource usage summary 

             Memory (GB)    NCPUs
```

Requested : 100 2
Used : 0 (peak) 0.61 (ave)

============================================_

Could you please help me with this?
In the output file above, is it normal to have 0 common ids? 

Many thanks.

**Comment by roshchupkin** [comment]
Hi, 
The first error was because you have files with different formats or sub-folders in your directory /$HOME/from_Sajjad/chr1/ folder  

Don't think that changing "as.matrix" to "values" would solve this problem. 

Regarding 0 common ids, it is not normal. Because of this you don't have results. Check please that genotype data, phenotype and covariates files have the same ids. Sometime it could be because ids in one file saved as float and in another file as string.  You need to change this.

**Comment by r-mustafa** [comment]
Hi,

Yes, for the first problem I realized that there was a txt file in the folder that is causing the problem. I moved that file and it was sorted.

Thank you for your suggestion, I will have a check on the IDs between those files.

The reason why I changed "as.matrix" into "values" was because in another submission of my job to the cluster I received another error messages saying:

/$HOME/hase-master/hdgwas/data.py:798: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
  self._data._data=df[df.columns[1:]].as_matrix()

Also, I downgraded my Python into version 2.7.15 as I found that the .py files could not run for the newer Python (3.6.7).
